srcdeps-core/src/main/java/org/srcdeps/core/config/ConfugurationOverrideVisitor.java

```diff
@@ -0,0 +1,372 @@
+/**
+ * Copyright 2015-2016 Maven Source Dependencies
+ * Plugin contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.srcdeps.core.config;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Stack;
+import java.util.StringTokenizer;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.srcdeps.core.BuildRequest.Verbosity;
+
+/**
+ * A {@link ConfigurationNodeVisitor} to override values in a {@link Configuration.Builder} by values coming from some
+ * higher-ranked sources, such as command line.
+ *
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+public class ConfugurationOverrideVisitor implements ConfigurationNodeVisitor {
+    private interface Deserializer {
+        Object deserialize(String value);
+    }
+
+    private static final Logger log = LoggerFactory.getLogger(ConfugurationOverrideVisitor.class);
+
+    /** {@link Map} from types we consider primitive to their respective {@link Deserializer}s */
+    private static final Map<Class<?>, Deserializer> primitiveTypes;
+
+    /** {@link Map} from Builder types we are able to traverse to their respective traversal methods */
+    private static final Map<Class<?>, Method> supportedSubbuilders;
+
+    static {
```

Comment:
I added JavaDoc to Deserializer interface https://github.com/srcdeps/srcdeps-core/pull/4/commits/0b3c319becd8f572a08e32673427607c219ee7ef#diff-06f2010cfb3fc6463bdaaa715f783c1fR50 and also added a comment to the static block: https://github.com/srcdeps/srcdeps-core/pull/4/commits/0b3c319becd8f572a08e32673427607c219ee7ef#diff-06f2010cfb3fc6463bdaaa715f783c1fR69 - Is that sufficient?